File: android/src/main/java/com/visioncameramlkit/domain/services/IImagePreprocessor.kt
package com.visioncameramlkit.domain.services

import com.visioncameramlkit.domain.models.ImagePreprocessingOptions
import com.visioncameramlkit.domain.models.ProcessedImage
import java.io.File

interface IImagePreprocessor {
  fun preprocessImage(
    imageFile: File,
    options: ImagePreprocessingOptions,
  ): ProcessedImage
}

File: ios/Domain/Services/IImagePreprocessor.swift
import Foundation

#if canImport(VisionCamera)
  import VisionCamera

  protocol IImagePreprocessor {
    func preprocessFrame(frame: any HybridFrameSpec, options: ImagePreprocessingOptions)
      -> ProcessedImage?

    func preprocessImage(imageFile: URL, options: ImagePreprocessingOptions)
      -> ProcessedImage?
  }
#endif
